packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -5776,6 +5776,122 @@ void main() {
     expect(state.currentTextEditingValue.text, '12345');
     expect(state.currentTextEditingValue.composing, TextRange.empty);
   });
+
+  group('callback errors', () {
+    testWidgets('onSelectionChanged can throw errors', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: EditableText(
+            readOnly: true,
+            controller: TextEditingController(text: 'Lorem ipsum dolor sit amet'),
+            backgroundCursorColor: Colors.grey,
+            focusNode: focusNode,
+            style: textStyle,
+            cursorColor: cursorColor,
+            selectionControls: materialTextSelectionControls,
+            onSelectionChanged: (TextSelection selection, SelectionChangedCause cause) {
+              throw Exception('Oh no');
+            },
+          ),
+        ),
+      );
+
+      // Interact with the field to establish the input connection.
+      final Offset topLeft = tester.getTopLeft(find.byType(EditableText));
+      await tester.tapAt(topLeft + const Offset(0.0, 5.0));
+      await tester.pump();
+
+      // Modify the selection and expect an exception.
+      final EditableTextState state = tester.state<EditableTextState>(find.byType(EditableText));
+      state.updateEditingValue(const TextEditingValue(
+        text: 'Foo bar',
+        selection: TextSelection(baseOffset: 0, extentOffset: 3),
+      ));
+      expect(tester.takeException(), isNotNull);
```

Comment:
There's an `isFlutterError` matcher, that seems appropriate here.

Bonus points for also checking that the exception message contains some of the text we are expecting here to ensure that this is actually the exception we want (and not some other random exception).

Reply:
(same below)